### PR TITLE
fix: set network telemetry client version on server

### LINKED ISSUE #14375: client_version not being set for networking telemetry
See conversion happening [here](https://github.com/coder/coder/blob/main/coderd/telemetry/telemetry.go#L1270)

### PR DESCRIPTION
**Pull request by ethanndickson**
Also removes the `DisconnectionReason` field, given that it was unused. Requires a corresponding change on the telemetry server.

Closes #14375

### CODE REVIEW

**Comment by ethanndickson** [comment]
* **#14376** <a href="https://app.graphite.dev/github/pr/coder/coder/14376?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ethanndickson and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: tailnet/proto/tailnet.proto [hunk 1]
```diff
@@ -172,7 +172,7 @@ message TelemetryEvent {
 	google.protobuf.Timestamp time = 2;
 	string application = 3;
 	Status status = 4;
-	string disconnection_reason = 5;
+	reserved 5;
```
Comment (ethanndickson) [inline tailnet/proto/tailnet.proto:175]: So the value doesn't get re-used.

**Review by deansheather — APPROVED** [review]

File: tailnet/proto/tailnet.proto (hunk 1 above)
Comment (deansheather) [reply tailnet/proto/tailnet.proto:175]: Please put a comment explaining why

**Review by sreya — COMMENTED** [review]
There aren't any tests to prevent this from happening in the future. Is there an opportunity to introduce some?

**Comment by ethanndickson** [comment]
I think it'd be beneficial to have an E2E network telemetry test. We'd just need to mock the telemetry batcher on a coderdtest instance so we can inspect the telemetry.

**Pull request merged**